src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -196,7 +196,7 @@ public void setEventTaskFilter(Predicate<? super EventTask> predicate) {
     @Override
     public synchronized void addDeadlineTask(DeadlineTask deadlineTask) {
         taskBook.addDeadlineTask(deadlineTask);
-        setDeadlineTaskFilter(null);
+        assert deadlineTask.isFinished() == false;
```

Comment:
Ugh, I just noticed that this code has actually been here since the start...

Hmmmm..... We will have to rethink this when we start implementing the filtering commands.